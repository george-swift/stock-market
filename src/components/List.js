import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({
  symbol,
  name,
  exchange,
  getUrl,
}) => (
  <li>
    <div className="company">
      <div>
        <p>{symbol}</p>
        <h4 className="my-1">
          <Link
            to={`/company/${symbol}`}
            onClick={() => getUrl(symbol)}
            replace
          >
            {name}
          </Link>
        </h4>
      </div>
      <small className="fst-italic text-white-50">{exchange}</small>
    </div>
  </li>
);

List.propTypes = {
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  exchange: PropTypes.string.isRequired,
  getUrl: PropTypes.func.isRequired,
};

export default List;
