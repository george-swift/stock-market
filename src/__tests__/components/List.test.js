import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import {
  render, within, screen, fireEvent,
} from '@testing-library/react';
import List from '../../components/List';

const symbol = 'MSFT';
const name = 'Microsoft';
const exchange = 'NASDAQ';
const getUrl = jest.fn((symbol) => symbol);

describe('Snapshot of the List component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <List symbol={symbol} name={name} exchange={exchange} getUrl={getUrl} />,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Querying the List component', () => {
  beforeEach(() => render(
    <List symbol={symbol} name={name} exchange={exchange} getUrl={getUrl} />,
  ));

  it('should have details for each provided company', () => {
    const companyInfo = document.querySelector('.company');

    const symbol = within(companyInfo).getByText('MSFT');
    expect(symbol).toBeInTheDocument();

    const name = within(companyInfo).getByText('MSFT').nextSibling;
    expect(name).toBeInTheDocument();
    expect(name).toHaveTextContent('Microsoft');

    const exchange = within(companyInfo).getByText('Microsoft').nextSibling;
    expect(exchange).toBeInTheDocument();
    expect(exchange).toHaveTextContent('NASDAQ');
  });

  it("should have a button to view the company's financial model", () => {
    const button = screen.getByRole('button');
    expect(button).toHaveProperty('type', 'button');
    expect(button).toHaveTextContent('See Profile');
    fireEvent.click(button);
    expect(getUrl).toHaveBeenCalledTimes(1);
  });
});
