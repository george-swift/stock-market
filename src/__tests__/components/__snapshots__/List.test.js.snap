// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the List component should render correctly 1`] = `
<li>
  <div
    className="company"
  >
    <p
      className="text-white-50"
    >
      MSFT
    </p>
    <h4
      className="my-1"
    >
      Microsoft
    </h4>
    <small
      className="fst-italic text-white-50"
    >
      NASDAQ
    </small>
  </div>
  <button
    className="btn btn-sm btn-outline-light"
    onClick={[Function]}
    type="button"
  >
    See Profile
  </button>
</li>
`;
